pom.xml

```diff
@@ -1482,5 +1447,121 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>spotbugs-on-java8+</id>
+      <activation>
+        <jdk>[1.8,)</jdk>
+      </activation>
+      <dependencyManagement>
+        <dependencies>
+          <dependency>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-annotations</artifactId>
+            <version>3.1.6</version>
+          </dependency>
+        </dependencies>
+      </dependencyManagement>
+
+      <dependencies>
+        <dependency>
+          <groupId>com.github.spotbugs</groupId>
+          <artifactId>spotbugs-annotations</artifactId>
+          <scope>provided</scope>
+          <optional>true</optional>
+        </dependency>
+      </dependencies>
+
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-maven-plugin</artifactId>
+            <version>3.1.6</version>
+            <configuration>
+              <failOnError>${findbugs.failOnError}</failOnError>
```

Comment:
the property should be renamed to match the spotbugs one `${spotbugs.failOnError}` - where it is set by other profiles (the mess that is `-DskipTests` for example then it should set this appropriately)